core/market/decentralized/src/matcher/matcher.rs

```diff
@@ -4,36 +4,37 @@ use std::sync::Arc;
 use thiserror::Error;
 use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender};
 
-use ya_client::model::market::{Demand, Offer, Proposal};
+use ya_client::model::market::Proposal;
 use ya_client::model::ErrorMessage;
 use ya_persistence::executor::DbExecutor;
 use ya_persistence::executor::Error as DbError;
+use ya_service_api_web::middleware::Identity;
 
 use crate::db::dao::*;
 use crate::db::models::Demand as ModelDemand;
 use crate::db::models::Offer as ModelOffer;
-use crate::db::models::{SubscriptionId, SubscriptionParseError};
+use crate::db::models::{SubscriptionId, SubscriptionParseError, SubscriptionValidationError};
 use crate::db::*;
 use crate::migrations;
 use crate::protocol::{
-    Discovery, DiscoveryError, DiscoveryInitError, Propagate, StopPropagateReason,
+    Discovery, DiscoveryError, DiscoveryInitError, ShouldPropagate, StopPropagateReason,
 };
 use crate::protocol::{OfferReceived, OfferUnsubscribed, RetrieveOffers};
 
 #[derive(Error, Debug)]
 pub enum DemandError {
-    #[error("Failed to save Demand. Error: {0}.")]
-    SaveDemandFailure(#[from] DbError),
+    #[error("Failed to save Demand [{1}]. Error: {0}.")]
+    SaveDemandFailure(DbError, SubscriptionId),
```

Comment:
We have spoken about this already...
This subscription id was generated while ago and tells nothing to provider/requestor.
No reason to print it.

Reply:
will revert this